src/traces/bar/hover.js

```diff
@@ -65,17 +65,18 @@ module.exports = function hoverPoints(pointData, xval, yval, hovermode) {
     if(Color.opacity(mc)) pointData.color = mc;
     else if(Color.opacity(mlc) && mlw) pointData.color = mlc;
 
+    var size = (trace.base) ? di.b + di.s : di.s;
```

Comment:
Ok. I think you got this right.

For reference, here's the non-stacked case under `base` restyle:

![gifrecord_2016-11-17_133959]
(https://cloud.githubusercontent.com/assets/6675409/20402731/eb1ee9a6-accb-11e6-8e7a-c06c88ccbda5.gif)

Perhaps we could set a special flag in the stacked bar calcdata traces [here](https://github.com/plotly/plotly.js/blob/564f661cd2831cb3e97a1eaec1d0f8505bedd2fb/src/traces/bar/set_positions.js#L96) instead of relying on the fact that all stacked bars have falsy `trace.base`. Or maybe adding a few test cases will suffice.